### ISSUE TITLE
Addition of a whitelist/blacklist of what items they can pick up

### ISSUE THREAD

**Issue by crafting-Aaron** [jaredlll08/SlimyBoyos#8]
Kind of makes sense they would be able to pick up their buddy's slime but since that does not make them stronger or anything it would be a better experience to restrict them from picking slimeballs so they can pick up something else.

Or who knows, maybe they _should_ be stronger if they pick up a slimeball.

Also maybe adding the trash and/or treasure loot table from fishing for them to spawn naturally with. Junk like leather boots, a bottle, a used iron sword, or shield adds a feeling of immersion that these slimes had a few fights and travels beforehand.

**Comment by ghost** [comment]
> Or who knows, maybe they _should_ be stronger if they pick up a slimeball.

I agree - I feel like if they pick up slimeballs they should absorb it and grow in size (with enough slimeballs, obviously).

**Comment by ghost** [comment]
> Also maybe adding the trash and/or treasure loot table from fishing for them to spawn naturally with. Junk like leather boots, a bottle, a used iron sword, or shield adds a feeling of immersion that these slimes had a few fights and travels beforehand.

This is also a super cool idea.

EDIT: In regards to implementing this I think it would just optionally pull from the loot table [here](https://github.com/jaredlll08/SlimyBoyos/blob/0a7d1862e77a02d652c4848a66db8413de7c71a6/Common/src/main/java/com/blamejared/slimyboyos/mixin/common/MixinLivingEntity.java#L96). Will investigate and write a PR when I have the chance.

**Comment by ghost** [comment]
@jaredlll08 Sorry to bother you - I plan to write a PR for some of the other stuff mentioned here and wanted to get some implementation details straightened out.

### Loot Tables Feature

Currently, I see that this mod doesn't utilise a config file, which so far works fine. However, if I were to implement the loot tables feature, I assume that it should be configurable (in regards to which loot tables, etc.), or even disableable, but this would require a config.

I am happy to set this up (including ModMenu / Catalogue entrypoints, etc.) but would like to know if that's something that is desired, or if you'd rather avoid it.

If you're happy with adding a config, is there a specific config library for each side you'd prefer to use?

### Slime Growth / Slimeball Item Consuming Feature

Once again this sort of thing would probably require a config to control the growth rate, etc.

However, I'd also like to understand what the desired growth rules would be:
- How many slime items does it take for a slime to grow to the next size? Is it dependent on the current side?
- Should the maximum size a slime can grow to be capped to the vanilla maximum ([_"Only sizes 1, 2, and 4 (NBT Size tag values 0, 1 and 3 respectively) spawn naturally."_](https://minecraft.fandom.com/wiki/Slime#:~:text=Only%20sizes%201%2C%202%2C%20and,Size%20tag%200%E2%80%93126))?
- Should a slime's health increase slowly for each individual slimeball, until it is ready to grow to the next size?
- Should a slime be capable of consuming slimeballs even if it is holding an item?
- Should a slime be capable of consuming slime blocks, and should this account for 9 slimeballs by default, or some smaller amount?

**Comment by jaredlll08** [comment]
The initial issue at hand here, an allowlist / blocklist has been resolved by https://github.com/jaredlll08/SlimyBoyos/pull/11

As for:
> Or who knows, maybe they should be stronger if they pick up a slimeball.

When I made this mod, the initial goal was just slimes picking up items and dropping them when they die, not spawning with them or consuming them.

I currently have no plans to expand it to include consuming items or spawning with items as that will affect gameplay and may not be clear to users who currently enjoy the mod as is.

@Brittank88 I would suggest that you should make your own mod that does this, that way you would have full control over how it should be done. Alternatively, you could also make an addon mod to this mod, adding those features.

If you end up making your own mod, be sure to let me know the modid so I can disable my mod's functionality if your mod is included, don't want slimes picking up two items at a time do we?

**Comment by ghost** [comment]
Understood! If anything, I'd make an add-on mod as you've already got a lot of the infrastructure set up.

**Comment by crafting-Aaron** [comment]
> 

Glad that I inspired the idea for this @Brittank88

 I suggested another idea in the Slimier Slimes mod that is similar to the size growth. However instead of picking up slimeballs it's about once smaller slimes notice a player, after a certain time and if they meet the count criteria, they will bounce towards each other and merge into the next size. 
In other words, 2 tiny (or was it 3-4?) Slimes after a couple seconds will jump towards each other and make a medium slime and so on. The merged slimes do not drop xp to avoid exploit. If there is a missing slime the merge won't happen. The medium slime or big slime will hold items the smaller ones had. So if you die while slimes are around you might come back to a surprise. Your gear is inside a big slime.

Why I came up with it? That mod spawns a group of tiny slimes on the surface or other places. But it looked too easy. So having an RPG or anime like event of slimes merging together would add some flavor to exploration.

I don't remember if the author of that mod agreed with the idea but if you are planning to make a similar addon, this was the original merged idea of both mods for me.
TLDR; overhauling slimes with absorbing and merging

**Comment by ghost** [comment]
@crafting-Aaron It's a very cool idea, though I would try and find a way to keep xp drops without introducing an exploit. Maybe I'll look into it after my exams are done!

**Comment by crafting-Aaron** [comment]
Good luck on your exams! Consult a tutor for any questions! Don't be like me just cramming the day before by myself.

**Comment by ghost** [comment]
> Good luck on your exams! Consult a tutor for any questions! Don't be like me just cramming the day before by myself.

Trying my best! Just a lot to revise... pthreads, OpenMP, MPI, CUDA... and that's just one of the two units.
